redaxo/src/core/fragments/core/navigation.php

```diff
@@ -1,4 +1,6 @@
-
+<?php
+/** @var rex_fragment $this */
```

Comment:
Alternativ würde wohl auch `assert($this instanceof rex_fragment);` funktionieren, was wir ja teils auch in anderen includeten Dateien für die Eingangsvariablen nutzen.
Ist mir hier allerdings eigentlich egal, welche Variante wir nehmen, solange sie funktioniert.

Man könnte auch überlegen, statt in allen Fragmenten das zu notieren, es per Plugin zu programmieren. Wobei das dann Tool-spezifisch wäre, und zum Beispiel auch von phpstorm vermutlich nicht verstanden werden würde. Daher bin ich auch einverstanden, wenn wir es in allen Fragmenten explizit mit einer der Varianten notieren.

Reply:
Funktioniert die assert variante auch mit phpstorm?

Da der nutzen von „this in fragmenten“ eher gering ist, würd ich da keine tool plugins o.ä. Für sehen

Reply:
> Funktioniert die assert variante auch mit phpstorm?

Habe es jetzt getestet, phpstorm würde hier beides verstehen. also sowohl `@var` funktioniert, als auch `assert()`.